src/persist/src/indexed/runtime.rs

```diff
@@ -999,12 +1023,29 @@ impl<L: Log, B: Blob> RuntimeImpl<L, B> {
 
         if need_step {
             self.prev_step = step_start;
-            if let Err(e) = self.indexed.step() {
-                self.metrics.cmd_step_error_count.inc();
-                // TODO: revisit whether we need to move this to a different log level
-                // depending on how spammy it ends up being. Alternatively, we
-                // may want to rate-limit our logging here.
-                log::warn!("error running step: {:?}", e);
+            let maintenance_reqs = match self.indexed.step() {
+                Ok(maintenance_reqs) => maintenance_reqs,
+                Err(e) => {
+                    self.metrics.cmd_step_error_count.inc();
+                    // TODO: revisit whether we need to move this to a different log level
+                    // depending on how spammy it ends up being. Alternatively, we
+                    // may want to rate-limit our logging here.
+                    log::warn!("error running step: {:?}", e);
+                    vec![]
+                }
+            };
+
+            for maintenance_req in maintenance_reqs {
+                let sender = self.tx.clone();
+                let maintenance_future = maintenance_req.to_future(&self.maintainer);
+                tokio::spawn(async move {
```

Comment:
let's thread an `Arc<tokio::runtime::Runtime>` here too to call this spawn on. I dislike the magic tokio thread local handle stuff, it's much too action at a distance. we have to use it in the blob stuff because the s3 library has bought into it, but it's avoidable here

Reply:
done. unfortunately I had to do this in the second commit 

```
commit ae82ae238796bb1ef9bb425fdcea3e13b3e5c3f7

    persist: add Maintainer to Runtime, sketch out MaintenanceRequests.
```

because that commit wouldn't run tests correctly without this change when rebased onto main, and it seemed much more challenging to rebase this successfully if i couldn't verify that each commit passed all tests as i went